src/formatDate.js

```diff
@@ -8,7 +8,86 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let result = '';
+  const oldDate = date;
+  const newFormat = toFormat;
+  let array = [];
+  let year = '';
+  let yearBig = '';
+  const newArray = [];
+  let indexYear = 0;
+  let indexYearbig = 0;
+
+  const [, , , oldSeparator] = fromFormat;
+  const [, , , newSeparator] = newFormat;
+
+  array = oldDate.split(oldSeparator);
+
+  if (newFormat.includes('YYYY') && fromFormat.includes('YYYY')) {
+    for (const n of newFormat) {
+      const searchInd = fromFormat.indexOf(n);
+
+      newArray.push(array[searchInd]);
+    }
+    newArray.pop();
+    result = newArray.join(newSeparator);
+
+    return result;
+  }
+
+  if (newFormat.includes('YY')) {
+    for (const n of array) {
+      if (n.length > 3) {
+        indexYear = array.indexOf(n);
+        year = n.slice(2);
+      }
+    }
+    array[indexYear] = year;
+
+    const newYY = newFormat.indexOf('YY');
+
+    newFormat[newYY] = 'YYYY';
+
+    for (const n of newFormat) {
+      const searchInd = fromFormat.indexOf(n);
+
+      newArray.push(array[searchInd]);
+    }
+    newArray.pop();
+
+    result = newArray.join(newSeparator);
+
+    return result;
+  }
```

Comment:
In this block, the function modifies `newFormat` directly by replacing 'YY' with 'YYYY'. This can lead to issues if `newFormat` is reused elsewhere. Consider creating a copy of `newFormat` before making modifications.